+ source ../common/common_lib.sh
++ source /root/mugen-riscv/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Sun Dec  4 04:30:43 2022 - INFO  - Start environmental preparation.
+ DNF_INSTALL proftpd
+ pkgs=proftpd
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs proftpd --node 1 --tempfile ''
+ tmpfile2='Last metadata expiration check: 0:00:24 ago on Sun 04 Dec 2022 04:30:58 AM EST.
Error: 
 Problem: conflicting requests
  - nothing provides libmemcached.so.11()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64
  - nothing provides libmemcachedutil.so.2()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64
(try to add '\''--skip-broken'\'' to skip uninstallable packages)'
+ '[' -z '' ']'
+ tmpfile='Last metadata expiration check: 0:00:24 ago on Sun 04 Dec 2022 04:30:58 AM EST.
Error: 
 Problem: conflicting requests
  - nothing provides libmemcached.so.11()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64
  - nothing provides libmemcachedutil.so.2()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64
(try to add '\''--skip-broken'\'' to skip uninstallable packages)'
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Sun Dec  4 04:31:24 2022 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Sun Dec  4 04:31:25 2022 - INFO  - Start testing...
+ test_execution proftpd.service
+ service=proftpd.service
++ date '+%Y-%m-%d %T'
+ log_time='2022-12-04 04:31:25'
+ test_restart proftpd.service
+ service=proftpd.service
+ systemctl restart proftpd.service
Failed to restart proftpd.service: Unit proftpd.service not found.
+ CHECK_RESULT 5 0 0 'proftpd.service restart failed'
+ actual_result=5
+ expect_result=0
+ mode=0
+ error_log='proftpd.service restart failed'
+ '[' -z 5 ']'
+ '[' 0 -eq 0 ']'
+ test 5x '!=' 0x
+ test -n 'proftpd.service restart failed'
+ LOG_ERROR 'proftpd.service restart failed'
+ message='proftpd.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'proftpd.service restart failed'
Sun Dec  4 04:31:26 2022 - ERROR - proftpd.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 34'
+ message='../common/common_lib.sh line 34'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 34'
Sun Dec  4 04:31:27 2022 - ERROR - ../common/common_lib.sh line 34
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status proftpd.service
+ grep 'Active: active'
Unit proftpd.service could not be found.
+ CHECK_RESULT 1 0 0 'proftpd.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='proftpd.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'proftpd.service restart failed'
+ LOG_ERROR 'proftpd.service restart failed'
+ message='proftpd.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'proftpd.service restart failed'
Sun Dec  4 04:31:34 2022 - ERROR - proftpd.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 37'
+ message='../common/common_lib.sh line 37'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 37'
Sun Dec  4 04:31:35 2022 - ERROR - ../common/common_lib.sh line 37
+ return 0
+ systemctl stop proftpd.service
Failed to stop proftpd.service: Unit proftpd.service not loaded.
+ CHECK_RESULT 5 0 0 'proftpd.service stop failed'
+ actual_result=5
+ expect_result=0
+ mode=0
+ error_log='proftpd.service stop failed'
+ '[' -z 5 ']'
+ '[' 0 -eq 0 ']'
+ test 5x '!=' 0x
+ test -n 'proftpd.service stop failed'
+ LOG_ERROR 'proftpd.service stop failed'
+ message='proftpd.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'proftpd.service stop failed'
Sun Dec  4 04:31:36 2022 - ERROR - proftpd.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 39'
+ message='../common/common_lib.sh line 39'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 39'
Sun Dec  4 04:31:37 2022 - ERROR - ../common/common_lib.sh line 39
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status proftpd.service
+ grep 'Active: inactive'
Unit proftpd.service could not be found.
+ CHECK_RESULT 1 0 0 'proftpd.service stop failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='proftpd.service stop failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'proftpd.service stop failed'
+ LOG_ERROR 'proftpd.service stop failed'
+ message='proftpd.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'proftpd.service stop failed'
Sun Dec  4 04:31:43 2022 - ERROR - proftpd.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 42'
+ message='../common/common_lib.sh line 42'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 42'
Sun Dec  4 04:31:44 2022 - ERROR - ../common/common_lib.sh line 42
+ return 0
+ systemctl start proftpd.service
Failed to start proftpd.service: Unit proftpd.service not found.
+ CHECK_RESULT 5 0 0 'proftpd.service start failed'
+ actual_result=5
+ expect_result=0
+ mode=0
+ error_log='proftpd.service start failed'
+ '[' -z 5 ']'
+ '[' 0 -eq 0 ']'
+ test 5x '!=' 0x
+ test -n 'proftpd.service start failed'
+ LOG_ERROR 'proftpd.service start failed'
+ message='proftpd.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'proftpd.service start failed'
Sun Dec  4 04:31:45 2022 - ERROR - proftpd.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 44'
+ message='../common/common_lib.sh line 44'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 44'
Sun Dec  4 04:31:46 2022 - ERROR - ../common/common_lib.sh line 44
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status proftpd.service
+ grep 'Active: active'
Unit proftpd.service could not be found.
+ CHECK_RESULT 1 0 0 'proftpd.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='proftpd.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'proftpd.service start failed'
+ LOG_ERROR 'proftpd.service start failed'
+ message='proftpd.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'proftpd.service start failed'
Sun Dec  4 04:31:53 2022 - ERROR - proftpd.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 47'
+ message='../common/common_lib.sh line 47'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 47'
Sun Dec  4 04:31:54 2022 - ERROR - ../common/common_lib.sh line 47
+ return 0
+ test_enabled proftpd.service
+ service=proftpd.service
++ systemctl is-enabled proftpd.service
Failed to get unit file state for proftpd.service: No such file or directory
+ state=
+ '[' '' == enabled ']'
+ '[' '' == disabled ']'
+ '[' '' == masked ']'
+ '[' '' == static ']'
+ LOG_INFO 'Unit is indirect, ignoring.'
+ message='Unit is indirect, ignoring.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Unit is indirect, ignoring.'
Sun Dec  4 04:31:55 2022 - INFO  - Unit is indirect, ignoring.
+ journalctl --since '2022-12-04 04:31:25' -u proftpd.service
+ grep -i 'fail\|error'
+ grep -v -i 'DEBUG\|INFO\|WARNING'
+ CHECK_RESULT 1 0 1 'There is an error message for the log of proftpd.service'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='There is an error message for the log of proftpd.service'
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ systemctl start proftpd.service
Failed to start proftpd.service: Unit proftpd.service not found.
+ sed -i 's\ExecStart = /usr/sbin/proftpd\ExecStart = /usr/sbin/proftpd -q\g' /usr/lib/systemd/system/proftpd.service
sed: can't read /usr/lib/systemd/system/proftpd.service: No such file or directory
+ systemctl daemon-reload
+ systemctl reload proftpd.service
Failed to reload proftpd.service: Unit proftpd.service not found.
+ CHECK_RESULT 5 0 0 'proftpd.service reload failed'
+ actual_result=5
+ expect_result=0
+ mode=0
+ error_log='proftpd.service reload failed'
+ '[' -z 5 ']'
+ '[' 0 -eq 0 ']'
+ test 5x '!=' 0x
+ test -n 'proftpd.service reload failed'
+ LOG_ERROR 'proftpd.service reload failed'
+ message='proftpd.service reload failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'proftpd.service reload failed'
Sun Dec  4 04:31:59 2022 - ERROR - proftpd.service reload failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_proftpd.sh line 35'
+ message='oe_test_service_proftpd.sh line 35'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_proftpd.sh line 35'
Sun Dec  4 04:32:00 2022 - ERROR - oe_test_service_proftpd.sh line 35
+ return 0
+ systemctl status proftpd.service
+ grep 'Active: active'
Unit proftpd.service could not be found.
+ CHECK_RESULT 1 0 0 'proftpd.service reload causes the service status to change'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='proftpd.service reload causes the service status to change'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'proftpd.service reload causes the service status to change'
+ LOG_ERROR 'proftpd.service reload causes the service status to change'
+ message='proftpd.service reload causes the service status to change'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'proftpd.service reload causes the service status to change'
Sun Dec  4 04:32:01 2022 - ERROR - proftpd.service reload causes the service status to change
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_proftpd.sh line 37'
+ message='oe_test_service_proftpd.sh line 37'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_proftpd.sh line 37'
Sun Dec  4 04:32:02 2022 - ERROR - oe_test_service_proftpd.sh line 37
+ return 0
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Sun Dec  4 04:32:03 2022 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 8
+ test 8 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Sun Dec  4 04:32:03 2022 - ERROR - failed to execute the case.
+ exit 8
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Sun Dec  4 04:32:04 2022 - INFO  - start environment cleanup.
+ sed -i 's\ExecStart = /usr/sbin/proftpd -q\ExecStart = /usr/sbin/proftpd\g' /usr/lib/systemd/system/proftpd.service
sed: can't read /usr/lib/systemd/system/proftpd.service: No such file or directory
+ systemctl daemon-reload
+ systemctl reload proftpd.service
Failed to reload proftpd.service: Unit proftpd.service not found.
+ systemctl stop proftpd.service
Failed to stop proftpd.service: Unit proftpd.service not loaded.
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z Last metadata expiration check: 0:00:24 ago on Sun 04 Dec 2022 04:30:58 AM EST.
Error: 
 Problem: conflicting requests
  - nothing provides libmemcached.so.11()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64
  - nothing provides libmemcachedutil.so.2()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64
(try to add '--skip-broken' to skip uninstallable packages) ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile 'Last metadata expiration check: 0:00:24 ago on Sun 04 Dec 2022 04:30:58 AM EST.
Error: 
 Problem: conflicting requests
  - nothing provides libmemcached.so.11()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64
  - nothing provides libmemcachedutil.so.2()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64
(try to add '\''--skip-broken'\'' to skip uninstallable packages)'
Traceback (most recent call last):
  File "/root/mugen-riscv/libs/locallibs/rpm_manage.py", line 181, in <module>
    exitcode = rpm_remove(args.node, args.pkgs, args.tempfile)
  File "/root/mugen-riscv/libs/locallibs/rpm_manage.py", line 152, in rpm_remove
    with open(tmpfile, "r") as f:
OSError: [Errno 36] File name too long: "Last metadata expiration check: 0:00:24 ago on Sun 04 Dec 2022 04:30:58 AM EST.\nError: \n Problem: conflicting requests\n  - nothing provides libmemcached.so.11()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64\n  - nothing provides libmemcachedutil.so.2()(64bit) needed by proftpd-1.3.7c-4.an8.riscv64\n(try to add '--skip-broken' to skip uninstallable packages)"
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Sun Dec  4 04:32:11 2022 - INFO  - Finish environment cleanup!
